### PR TITLE
RestClient throws an exception on 401

### PR DESCRIPTION
**Pull request by davidw**
https://github.com/rest-client/rest-client#exceptions-see-httpwwww3orgprotocolsrfc2616rfc2616-sec10html

We noticed a problem here:

https://www.pivotaltracker.com/n/projects/1282304/stories/133460509

### CODE REVIEW

File: lib/g5_authentication_client/auth_token_helper.rb [hunk 1]
```diff
@@ -2,15 +2,18 @@ module G5AuthenticationClient::AuthTokenHelper
   # Return response to 'yield'
   # Yield response should have a 'code' method for the http status code
   def do_with_username_pw_access_token
-    response = yield cached_username_pw_access_token
-    if 401 == response.code.to_i
```
Comment (bbauer) [inline lib/g5_authentication_client/auth_token_helper.rb]: Don't we still need this for apps that do not use RestClient?

Comment (davidw) [inline lib/g5_authentication_client/auth_token_helper.rb]: What else uses this?

Comment (bbauer) [inline lib/g5_authentication_client/auth_token_helper.rb]: I think pretty much all of our apps use this. So could be HTTParty or Net::HTTP etc...

**Comment by perryqh** [comment]
Most apps use httparty and the deleted lines are needed.

**Comment by davidw** [comment]
@perryqh I re-added those after the rescue, so that it now has 2 checks.  BTW, any ideas about the failing tests - as in they don't get run at all?

File: lib/g5_authentication_client/auth_token_helper.rb (hunk 1 above)
Comment (steveburkett) [inline lib/g5_authentication_client/auth_token_helper.rb]: this is tricky i think..i remember looking at this once.

**Pull request merged**